kamu-adapter-http/src/smart_protocol/dataset_helper.rs

```diff
@@ -487,10 +508,17 @@ pub async fn prepare_push_object_transfer_strategy(
                 expires_at: result.expires_at,
             }),
             Err(error) => match error {
-                GetTransferUrlError::NotSupported => Ok(TransferUrl {
-                    url: get_simple_transfer_protocol_url(object_file_ref, dataset_url),
-                    expires_at: None,
-                }),
+                GetTransferUrlError::NotSupported => {
+                    let mut simple_transfer_protocl_url =
+                        get_simple_transfer_protocol_url(object_file_ref, dataset_url);
+                    let staging_name_query_part = format!("staging_name={}", dataset_staging_name);
+                    simple_transfer_protocl_url.set_query(Some(staging_name_query_part.as_str()));
```

Comment:
Why do we need to carry over staging name between this phase and the PUT request?

How I see it:
- Client does "push"
- Server sees that the data object A is missing and sends `/data/A` URL
- Client sends `PUT /data/A` and streams the body
- Server (via `ObjectRepositoryFs`) streams into a temporary staging file, and only when stream completes and hash sum matches does atomic move `staging-xyz -> A`

Reply:
Discussed on Slack: key problem is that from perspective of PUT reuqest the target dataset does not yet exist as `DatasetBulder` is not finalized in another thread/process yet.

Reply:
Resolved